File: Jasmine/JasmineCompiler/Src/DFIR_Compiler/DFIR_Compiler/ITerminal.cs
using System;
using System.Collections.Generic;


namespace DFIR_Compiler
{
    public class ITerminal
    {
        private int terminalId;
        private int terminalIndex;
        private IDataType iDataType;
        private static IDictionary<int, ITerminal> IdToITerminal = new Dictionary<int, ITerminal>();
        private List<Connection> connections = new List<Connection>();

        public ITerminal(int TerminalId, int TerminalIndex, IDataType _iDataType)
        {
            terminalId = TerminalId;
            terminalIndex = TerminalIndex;
            iDataType = _iDataType;
            connections = new List<Connection>();
            IdToITerminal[terminalId] = this;
        }


        public virtual int GetITerminalId()
        {
            return terminalId;
        }
        public virtual int GetITerminalIndex()
        {
            return terminalIndex;
        }
        public virtual IDataType GetIDataType()
        {
            return iDataType;
        }
        public static IDictionary<int, ITerminal> GetIdToITerminal()
        {
            return IdToITerminal;
        }
        public virtual List<Connection> GetConnections()
        {
            return connections;
        }

       
        public virtual void AddConnection(Connection connection)
        {
            connections.Add(connection);
        }

        public virtual void Print()
        {
            Console.WriteLine("ITerminal:");
            Console.WriteLine("TerminalId: " + terminalId + ' ' + "TerminalIndex: " + terminalIndex);
            iDataType.Print();
            for (int i = 0; i < connections.Count; i++)
            {
                connections[i].Print();
            }
        }

    }
}

File: Wayne/WayneCompiler/src/Dfir_representation/ITerminal.java
package Dfir_representation;

import java.util.HashMap;
import java.util.Map;
import java.util.Vector;



public  class ITerminal{
	   
	   private int terminalId;
	   private int terminalIndex;
	   private IDataType iDataType;
	   private static Map<Integer, ITerminal> IdToITerminal = new HashMap<Integer, ITerminal>();
	   private Vector<Connection> connections = new Vector<Connection>();
	   
	   public ITerminal(int terminalId, int terminalIndex, IDataType iDataType){
		   this.terminalId = terminalId;
		   this.terminalIndex = terminalIndex;
		   this.iDataType = iDataType;
		   this.connections = new Vector<Connection>();
		   ITerminal.IdToITerminal.put(this.terminalId, this);
	   }
	   public IDataType getIDataType(){
		   return this.iDataType;
	   }
	   public int getTerminalId(){
		   return this.terminalId;
	   }
	   public int getTerminalIndex(){
		   return this.terminalIndex;
	   }
	   public static Map<Integer, ITerminal> getIdToITerminal(){
		   return ITerminal.IdToITerminal;
	   }
	   public Vector<Connection> getConnections(){
		   return this.connections;
	   }
	   public void AddConnection(Connection connection){
		   this.connections.addElement(connection);
	   }
	   public int getITerminalId(){
		   return this.terminalId;
	   }
	   public int getITerminalIndex(){
		   return this.terminalIndex;
	   }
	   public void print(){
		   System.out.println("ITerminal begin");
		   System.out.println("TerminalId: " + this.terminalId + ' ' + "TerminalIndex: " + this.terminalIndex);
		   iDataType.print();
		   for(int i=0; i<connections.size(); i++){
			   connections.get(i).print();
		   }
		   System.out.println("ITerminal end");
	   }
}
